src/HPolyhedron.jl

```diff
@@ -340,6 +345,108 @@ function tohrep(P::HPoly{N}) where {N}
     return P
 end
 
+"""
+    remove_redundant_constraints(P::PT;
+                                 backend=GLPKSolverLP()) where {N, HPoly{N}}
+
+Given a polyhedron in H-representation, return a new polyhedron with no reundant
+constraints.
+
+## Input
+
+- `P`       -- polyhedron
+- `backend` -- (optional, default: `GLPKSolverLP`) the numeric LP solver backend
+
+## Output
+
+The polyhedron obtained by removing the redundant constraints in `P`.
+
+## Algorithm
+
+See `remove_redundant_constraints!`. 
+"""
+function remove_redundant_constraints(P::PT;
+                                      backend=GLPKSolverLP()) where {N, PT<:HPoly{N}}
+    return remove_redundant_constraints!(copy(P), backend=backend)
+end
+
+"""
+    remove_redundant_constraints!(P::PT;
+                                  backend=GLPKSolverLP()) where {N, HPoly{N}}
+
+Remove the reundant constraints in a polyhedron in H-representation; the polyhedron
+is updated inplace.
+
+## Input
+
+- `P`       -- polyhedron
+- `backend` -- (optional, default: `GLPKSolverLP`) the numeric LP solver backend
+
+## Output
+
+The polyhedron obtained by removing the redundant constraints in `P`.
+
+## Algorithm
+
+If the polyhedron `P` has `m` constraints and its dimension is `n`,
+this function checks one by one if each of the `m` constraints is
+implied by the `m-1` others. To check if the `k`-th constraint
+is redundant, an LP is formulated.
+
+For details, see [Fukuda's Polyhedra
+FAQ](https://www.cs.mcgill.ca/~fukuda/soft/polyfaq/node24.html).
+"""
+function remove_redundant_constraints!(P::PT;
+                                       backend=GLPKSolverLP()) where {N, PT<:HPoly{N}}
+
+    A, b = tosimplehrep(P)
+    m, n = size(A)
+    non_redundant_indices = 1:m
+
+    i = 1 # counter over reduced constraints
+    j = 1 # counter over original constraints
+
+    while j <= m
```

Comment:
Can we use a `for` loop here? `j` seems to be incremented every round.

Reply:
yes, we can use a `for` loop, i can change that.